api/views.py

```diff
@@ -0,0 +1,331 @@
+        return user in course.students.all() or user in course.teachers.all() \
+            or user == course.creator
+
+    def post(self, request, format=None):
+        try:
+            questionpaperid = request.data['question_paper']
+            attempt_number = request.data['attempt_number']
+            course_id = request.data['course']
+        except KeyError:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+        user = request.user
+        ip = request.META['REMOTE_ADDR']
+        questionpaper = self.get_questionpaper(questionpaperid)
+        course = self.get_course(course_id)
+        if not self.is_user_allowed(user, course):
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+        answerpaper = questionpaper.make_answerpaper(user, ip, attempt_number,
+                                                     course_id)
+        serializer = AnswerPaperSerializer(answerpaper)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+class AnswerValidator(APIView):
+
+    def get_answerpaper(self, pk, user):
+        try:
+            return AnswerPaper.objects.get(pk=pk, user=user)
+        except AnswerPaper.DoesNotExist:
+            raise Http404
+
+    def get_question(self, pk, answerpaper):
+        try:
+            question = Question.objects.get(pk=pk)
+            if question in answerpaper.questions.all():
+                return question
+            else:
+                raise Http404
+        except AnswerPaper.DoesNotExist:
+            raise Http404
+
+    def get_answer(self, pk):
+        try:
+            return Answer.objects.get(pk=pk)
+        except Answer.DoesNotExist:
+            raise Http404
+
+    def post(self, request, answerpaper_id, question_id, format=None):
+        try:
+            user_answer = request.data['answer']
+        except KeyError:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+        user = request.user
+        answerpaper = self.get_answerpaper(answerpaper_id, user)
+        question = self.get_question(question_id, answerpaper)
+        # save answer uid
+        answer = Answer.objects.create(question=question, answer=user_answer)
+        answerpaper.answers.add(answer)
+        answerpaper.save()
+        json_data = None
+        if question.type in ['code', 'upload']:
+            json_data = question.consolidate_answer_data(user_answer, user)
+        result = answerpaper.validate_answer(user_answer, question, json_data,
+                                             answer.id)
+
+        # updaTE RESult
+        if question.type not in ['code', 'upload']:
+            if result.get('success'):
+                answer.correct = True
+                answer.marks = question.points
+                answer.error = json.dumps(result.get('error'))
+                answer.save()
+            answerpaper.update_marks(state='inprogress')
+        return Response(result)
+
+    def get(self, request, uid):
```

Comment:
Both `get` and `post` methods are using the same code defined in `yaksh.views`. Perhaps, you can use it. Or atleast break it down into separate functions in `yaksh.views` so that you can use it here.